cunumeric/_ufunc/ufunc.py

```diff
@@ -325,19 +498,24 @@ def _find_common_type(arrs, orig_args):
         return np.find_common_type(array_types, scalar_types)
 
     def _resolve_dtype(self, arrs, orig_args, casting, precision_fixed):
-        common_dtype = self._find_common_type(arrs, orig_args)
+        if self._use_common_type:
+            common_dtype = self._find_common_type(arrs, orig_args)
+            key = (common_dtype.char, common_dtype.char)
+        else:
+            key = tuple(arr.dtype for arr in arrs)
 
-        key = (common_dtype.char, common_dtype.char)
         if key in self._types:
             arrs = [arr.astype(common_dtype) for arr in arrs]
```

Comment:
`common_dtype` is not necessarily defined here

Reply:
9b6d999b8b2e1a99e55179a2de117928d067c3f9